odoo/exceptions.py

```diff
@@ -107,3 +114,53 @@ def __init__(self, msg, tb):
 
 class QWebException(Exception):
     pass
+
+
+class ViewError(except_orm, ValueError):
+
+    def __init__(self, msg, view):
+        not_avail = _('n/a')
+        message = (
+            "%(msg)s\n\n" +
+            _("Error context:\nView `%(view_name)s`") +
+            "\n[view_id: %(viewid)s, xml_id: %(xmlid)s, "
+            "model: %(model)s, parent_id: %(parent)s]"
+        ) % {
+            'view_name': view.name or not_avail,
+            'viewid': view.id or not_avail,
+            'xmlid': view.xml_id or not_avail,
+            'model': view.model or not_avail,
+            'parent': view.inherit_id.id or not_avail,
+            'msg': msg,
+        }
+        vals = [self, message, '']
```

Comment:
Putting all that data in the error message is useless if the error provides a direct link to the view.